src/components/Upload.tsx

```diff
@@ -53,6 +53,41 @@ export default function Upload() {
 
   const onDrop = useCallback(
     (acceptedFiles: File[], rejectedFiles: FileRejection[]) => {
+      const parse = (file: File): boolean => {
+        file.arrayBuffer().then((arrayBuffer) => {
+          const byteArray = new Uint8Array(arrayBuffer);
+          try {
+            const dataSet = dicomParser.parseDicom(byteArray /*, options */);
+            const studyInstanceUid = dataSet.string("x0020000d");
+            if (studyUid.current !== studyInstanceUid) {
+              if (studyUid.current === "") studyUid.current = studyInstanceUid;
+              else {
+                alert(
+                  "Files studyInstanceUid doesn't match. Files will be reset!"
+                );
+                dispatch({ type: ActionType.RESET });
+                studyUid.current = "";
+                return false;
+              }
+            }
```

Comment:
ja bym trochę to zmodyfikował
```js
if (studyUid.current === "" && studyUid.current !== studyInstanceUid) {
    studyUid.current = studyInstanceUid
} else {
alert(
                  "Files studyInstanceUid doesn't match. Files will be reset!"
                );
                dispatch({ type: ActionType.RESET });
                studyUid.current = "";
                return false;
} 
```
To jest to samo, a ma mniej zagnieździeń i dzięki temu jest czytelniejsze